alerts/openshift-container-storage-operator/StorageAutoScalingFailed.md

```diff
@@ -0,0 +1,75 @@
+# StorageAutoScalingFailed
+
+## Meaning
+
+During automatic storage scaling, failures may occur due to various factors. In
+such instances, the `StorageAutoScalingFailed` alert is fired.
+
+## Impact
+
+The storage cannot be scaled at the moment.
+
+## Diagnosis
+
+There are multiple stages where the storage configurations might not have matched
+the requested storage parameters.
+
+1. OSD pod is in `Pending` state:
+    If it is in `Pending` state with the following logs, there is a high chance
+    it is due to the exhaustion of the resources.
+    ```bash
+    Warning  FailedScheduling  3m53s  default-scheduler  0/6 nodes are available: 1 Insufficient cpu, 2 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/6 nodes are available: 1 Insufficient cpu, 5 Preemption is not helpful for scheduling.
+    ```
+
+    Describe the pod in `Pending` state and check for the events.
+    ```bash
+    oc describe pod <osd_pod> -n <namespace>
+    ```
+
+    If there are no errors, retrieve the pvc of the pod in `Pending` state.
+    ```bash
+    pvcName=$(oc get pod <osd_pod> -n <namespace> -o jsonpath='{.spec.volumes[*].persistentVolumeClaim.claimName}')
+    oc get pvc $pvcName -n <namespace> -o yaml
+    ```
+    Check for the health and events of the PVC for detailed information.
+
+2. PVC is in `Pending` state:
+
+    ```bash
+    oc get pvc <pvc_name> -n <namespace> -o yaml
+    ```
+
+    Inspect the events of the pvc. Additionally, retrieve the pv of the pvc if
+    it is still in `pending` state.
+    ```bash
+    oc get pvc <pvc_name> -n <namespace> -o jsonpath='{.spec.volumeName}'
+    ```
+
+3. PV is in `Failed` state:
+    ```bash
+    oc get pv <pv_name> -n <namespace> -o yaml
+    ```
+    If the pv is in `Failed` state, there might be a storage issue. Check the
+    events of the pv for more information.
+
+## Mitigation
+
+1. If resources are depleted and pods start failing, allocate additional
+resources to the storage cluster. Possible solutions include adding more nodes
+or scaling up CPU and memory on existing nodes.
+- For instructions on adding nodes, please refer to this document: [Adding a node](https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.17/html-single/scaling_storage/index#adding_a_node)
+
+2. Check upon the Ceph Storage: In case the OSDs are `down` or `out`, restart them
+and try again.
+
+3. During vertical scaling, if the pv takes time to resize, then the osds would not
```

Comment:
@parth-gr I don't think restarting the rook operator will help. Restarting the OSD pods sometimes can help nudge the resize to completion though.

Reply:
Restarting the operator helps in the vertical scaling, When we need to update the crush-weight, which is not updated by `bluefs-bdev-expand` cmd and rook operator separately does that during osd upgrades.

that's correct it doesn't help in osd to get resize, but it helps in correcting the crush-weight if osd vertical scaling was manually triggered by re-starting the osd pod

@ShravaniVangur can you frame in that way, if we are doing osd pod restart better to also restart operator after that

Reply:
@parth-gr The only reason restarting the operator would help is if the Rook operator got stuck reconciling. So you're suggesting to restart the operator just in case it got stuck? That seems like a separate mitigation from restarting the OSDs, maybe it should be a separate bullet point?

Reply:
> So you're suggesting to restart the operator just in case it got stuck?

Even if it's not stuck, Rook operator does have a watch on when the osd is resize or re-started.
The bug: https://issues.redhat.com/browse/DFBUGS-965